### PR TITLE
Fixes #5824 - compare-object example

### LINKED ISSUE #5824: a more common compare two sets of objects example
I'm surprised there's not already an example like this comparing two sets of objects.  This is a common point of confusion .  You usually need to specify -property when comparing objects.
```powershell
$a = [pscustomobject]@{name='joe'}
$b = [pscustomobject]@{name='joey'}

compare-object $a $b  # no output

compare-object $a $b -property name

name SideIndicator
---- -------------
joey =>
joe  <=
```

Actually, I don't even think this is true:

> The result of the comparison indicates whether a property value appeared only in the reference object (<=) or only in the difference object (=>). If the IncludeEqual parameter is used, (==) indicates the value is in both objects.

```powershell
$a = [pscustomobject]@{name='joe';address='here'}   
$b = [pscustomobject]@{name='joey'}              
compare-object $a $b  # no output
```

It looks to me that example 4 is just converting each array to strings, and then comparing the string versions.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 75188614-c25b-02bf-f3bc-da5ab0aa1e30
* Version Independent ID: 85f1e969-8216-bed4-2204-409cbd9befd4
* Content: [Compare-Object (Microsoft.PowerShell.Utility)](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/compare-object?view=powershell-7#feedback)
* Content Source: [reference/7.0/Microsoft.PowerShell.Utility/Compare-Object.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.0/Microsoft.PowerShell.Utility/Compare-Object.md)
* Product: **powershell**
* GitHub Login: @JamesWTruher
* Microsoft Alias: **jimtru**

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#1711226](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1711226) - Fixes #5824 - compare-object example

## PR Context

Select the type(s) of documents being changed.

**Cmdlet reference & about_ topics**
- [x] Version 7.x preview content
- [x] Version 7.0 content
- [x] Version 6 content
- [x] Version 5.1 content

**Conceptual articles**
- [ ] Fundamental conceptual articles
- [ ] Script sample articles
- [ ] DSC articles
- [ ] Gallery articles
- [ ] JEA articles
- [ ] WMF articles
- [ ] SDK articles

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md [hunk 1]
```diff
@@ -36,6 +36,11 @@ used, (`==`) indicates the value is in both objects.
 If the **reference** or the **difference** objects are null (`$null`), `Compare-Object` generates a
 terminating error.
 
+For simple objects, like strings or numbers, the cmdlet compares the values of the objects. For
+complex objects, you must provide one or more properties to be used for comparison. If you do
+provide a property, the cmdlet checks for **IComparable** interfaces to perform the comparison. If
+that fails, the object are converted to strings and the string values are compared.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:42]: `object` should be plural or changed to 'each object is converted...'

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md [hunk 2]
```diff
@@ -216,6 +222,39 @@ output displayed by the default format for **System.Boolean** objects didn't dis
 **SideIndicator** property. However, the returned **System.Boolean** object has the added
 **NoteProperty**.
 
+### Example 6 - Comparing complex objects
+
+This example show the behavior when comparing complex objects. In this example we store two
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:227]: `show` should be plural.

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md [hunk 3]
```diff
@@ -216,6 +222,39 @@ output displayed by the default format for **System.Boolean** objects didn't dis
...
+different process objects for different instances of PowerShell. Both variables contain process
+object with the same name. When the objects are compared without specifying the **Property**
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:229]: `objects` should be plural or changed to 'contain a process object'.

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md [hunk 4]
```diff
@@ -216,6 +222,39 @@ output displayed by the default format for **System.Boolean** objects didn't dis
...
+parameter, the cmdlet consider the objects to be equal.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:230]: `consider` should be plural.

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md [hunk 5]
```diff
@@ -216,6 +222,39 @@ output displayed by the default format for **System.Boolean** objects didn't dis
...
+
+PS> Compare-Object $a $b -Property ProcessName, Id, CPU
+
+ProcessName    Id       CPU SideIndicator
+-----------    --       --- -------------
+pwsh        17600   11.4375 =>
+pwsh        11168 36.203125 <=
+```
+
+When you specify properties to be compared, the cmdlet show the differences.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:256]: `show` should be plural.

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md (hunk 1 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:42]: Fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:227]: Fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md (hunk 3 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:229]: Fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md (hunk 4 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:230]: Fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md (hunk 5 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md:256]: Fixed

**Pull request merged**